lib/galaxy/model/__init__.py

```diff
@@ -1510,7 +1510,13 @@ def __init__( self, history, action, role ):
         self.role = role
 
 
-class Dataset( object ):
+class StorableObject( object ):
+
+    def __init__( self, id, **kwargs):
+        self.id = id
```

Comment:
Am I missing any other Galaxy objects that get passed into object stores?

For my recordkeeping, this is how I found MetadataFile: `git grep update_from_file` lead me to https://github.com/mr-c/galaxy/blob/storableobject/lib/galaxy/datatypes/metadata.py#L603